### ISSUE TITLE
[stable/external-dns] How to setup AWS Route53 using preexisting IAM user secrets?

### ISSUE THREAD

**Issue by feliperfmarques** [helm/charts#13292]
**Is this a request for help?**:

Hello guys,

I can see that PR #8587 changes extraEnv to be an array of objects. It makes external-dns be able to retrieve credentials from already existing secrets, avoiding duplication and more importantly credentials stored in helm overrides which might be checked into version control. 

But, I can't see in docs how I should do this for config aws route53 using ExtraEnv to retrieve credentials from already existing secrets.  How should I do this?

**Version of Helm and Kubernetes**:
Helm - v2.13.1
Kubernetes - v1.11.9

**Which chart**: 

estable/external-dns

**What you expected to happen**:

Setup in AWS Route53 using preexisting IAM user secrets.

**Comment by feliperfmarques** [comment]
Hello guys,

I figured out how to do. It was stupid of me. It's very simple. But my doubt occurs because in docs din't have full description about this. The way like works with configs defined directly set up by  ```aws.secretKey``` and ```aws.accessKey``` in values.yaml and secrets defined by ValueFrom in ```ExtraEnvs``` are differents:

1 - When set  ```aws. secretKey``` and ```aws.accessKey``` this Chart stores the infos in an secret of the name provide by value of ```{{ external-dns.fullname }}``` and mount this secret in file of the name ```config``` in ```credentialsPath``` defined in values.yaml (pre external-dns 0.5.9 home dir should be `/root/.aws`), so AWS CLI is  loaded from a profile in the configuration file. But as mentioned, this approach genereted duplication and more importantly credentials stored in helm configs might be checked into version control.

2 - As documented by Amazon, environment variables provide another way to specify configuration options and credentials. Important note: By Precedence of options, It overrides any value loaded from a profile in the configuration file, that is, the values defined in ```aws. secretKey``` and ```aws.accessKey``` will be disregarded in favor ```AWS_ACCESS_KEY_ID``` and ```AWS_SECRET_ACCESS_KEY``` in the case defined both ways. . (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)

So basically, when defined var envs ```AWS_ACCESS_KEY_ID``` and ```AWS_SECRET_ACCESS_KEY``` the AWS CLI is  loaded from a profile defined in the var envs. Below, an example of configuration setting var envs ```AWS_ACCESS_KEY_ID``` and  ```AWS_SECRET_ACCESS_KEY``` with an secret named ```existing-secret``` which conatins keys ```aws_access_key_id``` and ```aws_secret_access_key```:

```
 - name: AWS_ACCESS_KEY_ID
   valueFrom:
     secretKeyRef:
       name: existing-secret
       key: aws_access_key_id

 - name: AWS_SECRET_ACCESS_KEY 
   valueFrom:
     secretKeyRef:
       name: existing-secret
       key: aws_secret_access_key 
```

**Comment by feliperfmarques** [comment]
PR #13336 submitted for improving docs.
